server/src/test/java/io/crate/role/RolePropertiesTest.java

```diff
@@ -117,4 +124,21 @@ public void test_bwc_streaming() throws Exception {
         assertThat(actual.jwtProperties()).isNull();
         assertThat(actual.sessionSettings()).isEqualTo(Map.of());
     }
+
+    @Test
+    public void test_can_read_empty_role_properties_from_x_content() throws IOException {
+        XContentBuilder xContentBuilder = JsonXContent.builder();
+        xContentBuilder.startObject();
+        xContentBuilder.endObject();
+
+        XContentParser parser = JsonXContent.JSON_XCONTENT.createParser(
+            NamedXContentRegistry.EMPTY,
+            DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+            Strings.toString(xContentBuilder));
+        Role.Properties properties = Role.Properties.fromXContent(parser);
+
+        assertThat(properties.jwtProperties()).isNull();
+        assertThat(properties.password()).isNull();
+        assertThat(properties.sessionSettings()).isNotNull(); // Must be always not null as Sessions.newSession doesn't do a NULL check.
```

Comment:
I don't think we need to comment on basic constraints and usage expectations.
Also, why not expect what we want?
```suggestion
        assertThat(properties.sessionSettings()).isEqualTo(Map.of()); 
```